lib/vanagon/driver.rb

```diff
@@ -23,16 +23,22 @@ def initialize(platform, project, options = { :configdir => nil, :target => nil,
 
       @platform = Vanagon::Platform.load_platform(platform, File.join(@@configdir, "platforms"))
       @project = Vanagon::Project.load_project(project, File.join(@@configdir, "projects"), @platform, components)
+      @@logger = Logger.new('vanagon_hosts.log')
+      @@logger.progname = 'vanagon'
 
       # If a target has been given, we don't want to make any assumptions about how to tear it down.
       engine = 'base' if target
+      # Hardware has explicit teardown to unlock the node
+      engine = 'hardware' if @platform.build_host
       require "vanagon/engine/#{engine}"
-      @engine = Object.const_get("Vanagon::Engine::#{engine.capitalize}").new(@platform, target)
+      if engine == 'hardware'
```

Comment:
You shouldn't need this conditional. If you look at what the pooler engine does, it takes 2 parameters to the constructor and just ignores the second.